system/src/ble_provisioning_mode_handler.cpp

```diff
@@ -60,8 +77,8 @@ int BleListeningModeHandler::constructControlRequestAdvData() {
     CHECK_TRUE(tempAdvData.append(BLE_SIG_AD_TYPE_FLAGS), SYSTEM_ERROR_NO_MEMORY);
     CHECK_TRUE(tempAdvData.append(BLE_SIG_ADV_FLAGS_LE_ONLY_GENERAL_DISC_MODE), SYSTEM_ERROR_NO_MEMORY);
 
-    char devName[32] = {};
-    CHECK(get_device_name(devName, sizeof(devName)));
+    char devName[BLE_MAX_DEV_NAME_LEN] = {};
```

Comment:
Should probably be `BLE_MAX_DEV_NAME_LEN + 1`. Additional care probably needs to be taken that it's null-terminated (in case returned name for some reason is > buffer size), see https://github.com/particle-iot/device-os/pull/2382/files#diff-522277e93ac030d207929e5d20501837ba30f5d94daaf8759714f2910cec970dR84